File: src/main/java/hu/gerviba/webschop/model/ItemCategory.java
package hu.gerviba.webschop.model;

import lombok.AllArgsConstructor;
import lombok.Getter;

@AllArgsConstructor
public enum ItemCategory {
    DEFAULT(0),
    ALPHA(1),
    BETA(2),
    GAMMA(3),
    DELTA(4),
    LAMBDA(5);

    @Getter
    final int id;

    public static ItemCategory of(Integer category) {
        for (ItemCategory value : values()) {
            if (value.getId() == category)
                return value;
        }
        return DEFAULT;
    }
}

File: src/main/kotlin/hu/kirdev/schpincer/model/ItemCategory.kt
package hu.kirdev.schpincer.model

enum class ItemCategory(val id: Int) {
    DEFAULT(0),
    ALPHA(1),
    BETA(2),
    GAMMA(3),
    DELTA(4),
    LAMBDA(5);

    companion object {
        fun of(category: Int): ItemCategory {
            for (value in values()) {
                if (value.id == category)
                    return value
            }
            return DEFAULT
        }
    }

}